src/strategy/strategies/softcodedprice/softcodedprice.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * @description Soft Coded Price
+ * @author @tsrnc2
+ *
+ * Limit orders without tying up funds with stop loss calculated on period close
+ * @todo when ordersize can be set allow, for multiple sell points to reduce risk
+ */
+import { PeriodItem } from '@lib'
+import { BaseStrategy, StrategyField } from '../base-strategy'
+import { Signal } from '@m8bTypes'
+import { logger } from '../../../util/logger'
+
+export interface SoftCodedPriceOptions {
+  period: string
+  buyPrice: number
+  stopLimit: number
+  sellPrice: number
+  isRepeat: boolean
+  }
+
+export class SoftCodedPrice extends BaseStrategy<SoftCodedPriceOptions> {
+  public static description =
+  'Buy in and sell a at specific price with a stop limit. Optionaly repeatable'
+
+  public static fields: StrategyField[] = [
+    {
+      name: 'period',
+      type: 'text',
+      prettyName: 'Period Length',
+      description: 'Length of time to sample.',
+      default: '1m',
+    },
+    {
+      name: 'buyPrice',
+      type: 'number',
+      prettyName: 'Buy Price',
+      description: 'If the price is at or bellow this price buy.',
+      default: 0,
+    },
+    {
+      name: 'stopLimit',
+      type: 'number',
+      prettyName: 'Stop Limit',
+      description: 'If after a buy the price gets to this point sell.',
+      default: 0,
+    },
+    {
+      name: 'sellPrice',
```

Comment:
Maybe use `stopLoss` as this name?

Also, the description for `buyPrice` and this field are too similar. Maybe use `up` and `down` in the descriptions like:

`Sell if the price goes up to this point`
`Sell if the price goes down to this point`